src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -401,6 +414,37 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
             }
         }
 
+#nullable enable
+        private void RemoveCallingConventionMismatches<TMember>(ArrayBuilder<MemberResolutionResult<TMember>> results, Cci.CallingConvention expectedConvention) where TMember : Symbol
+        {
+            if (typeof(TMember) != typeof(MethodSymbol))
+            {
+                return;
+            }
+
+            for (int i = 0; i < results.Count; i++)
+            {
+                var result = results[i];
+                var member = (MethodSymbol)(Symbol)result.Member;
+                if (result.Result.IsValid)
+                {
+                    if (!member.CallingConvention.IsCallingConvention(expectedConvention))
+                    {
+                        results[i] = new MemberResolutionResult<TMember>(
+                            result.Member, result.LeastOverriddenMember,
+                            MemberAnalysisResult.WrongCallingConvention());
+                    }
+                    else if (member.CallingConvention.HasUnknownCallingConventionAttributeBits())
+                    {
+                        results[i] = new MemberResolutionResult<TMember>(
```

Comment:
>results[i] = new MemberResolutionResult<TMember>( [](start = 24, length = 49)

Do we have a test covering this code path and asserting diagnostics that we report?